src/main/java/seedu/edulog/ui/StudentCard.java

```diff
@@ -64,6 +64,8 @@ public StudentCard(Student student, int displayedIndex) {
         student.getTags().stream()
                 .sorted(Comparator.comparing(tag -> tag.tagName))
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+        assert(paidIcon != null);
```

Comment:
yep, this is a good  assertion check. I suppose you could also assert that all the students are not null here as well. Looks good :)